src/betterproto/__init__.py

```diff
@@ -343,20 +343,43 @@ def _pack_fmt(proto_type: str) -> str:
     }[proto_type]
 
 
-def encode_varint(value: int) -> bytes:
-    """Encodes a single varint value for serialization."""
-    b: List[int] = []
-
-    if value < 0:
+def dump_varint(value: int, stream: BinaryIO) -> None:
+    """Encodes a single varint and dumps it into the provided stream."""
+    if value < -(1 << 63):
+        raise ValueError(
+            "Negative value is not representable as a 64-bit integer - unable to encode a varint within 10 bytes."
+        )
+    elif value < 0:
         value += 1 << 64
 
     bits = value & 0x7F
     value >>= 7
     while value:
-        b.append(0x80 | bits)
+        stream.write((0x80 | bits).to_bytes(length=1, byteorder="little"))
```

Comment:
```suggestion
        stream.write((0x80 | bits).to_bytes(1, "little"))
```
kwargs here are gonna hurt perf and they don't really add much